src/Compilers/Core/Portable/CodeGen/ItemTokenMap.cs

```diff
@@ -71,7 +68,7 @@ public IEnumerable<string> GetAllItems()
 
         //TODO: why is this is called twice during emit?
         //      should probably return ROA instead of IE and cache that in Module. (and no need to return count)
-        public IEnumerable<string> GetAllItemsAndCount(out int count)
+        public IEnumerable<T> GetAllItemsAndCount(out int count)
```

Comment:
Consider returning `ReadOnlyArray<T>` or `ImmutableArray<T>` now, even without caching in `Module`. We're already allocating an array and returning an array type would avoid confusion at the caller where we currently have an `IEnumerable<T>` from a mutable object.